internal/core/data/v2/controller/http/event_test.go

```diff
@@ -173,28 +173,30 @@ func TestAddEvent(t *testing.T) {
+		{"Invalid - No Event", []requests.AddEventRequest{noEvent}, "", "", true, http.StatusBadRequest},
+		{"Invalid - No Event Id", []requests.AddEventRequest{noEventID}, noEventID.Event.ProfileName, noEventID.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - Bad Event Id", []requests.AddEventRequest{badEventID}, badEventID.Event.ProfileName, badEventID.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Event DeviceName", []requests.AddEventRequest{noEventDevice}, noEventDevice.Event.ProfileName, noEventDevice.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Event ProfileName", []requests.AddEventRequest{noEventProfile}, noEventProfile.Event.ProfileName, noEventProfile.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Event Origin", []requests.AddEventRequest{noEventOrigin}, noEventOrigin.Event.ProfileName, noEventOrigin.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading", []requests.AddEventRequest{noReading}, noReading.Event.ProfileName, noReading.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading DeviceName", []requests.AddEventRequest{noReadingDevice}, noReadingDevice.Event.ProfileName, noReadingDevice.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading ResourceName", []requests.AddEventRequest{noReadingResourceName}, noReadingResourceName.Event.ProfileName, noReadingResourceName.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading ProfileName", []requests.AddEventRequest{noReadingProfileName}, noReadingProfileName.Event.ProfileName, noReadingProfileName.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading Origin", []requests.AddEventRequest{noReadingOrigin}, noReadingOrigin.Event.ProfileName, noReadingOrigin.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No Reading ValueType", []requests.AddEventRequest{noReadingValueType}, noReadingValueType.Event.ProfileName, noReadingValueType.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - Invalid Reading ValueType", []requests.AddEventRequest{invalidReadingInvalidValueType}, invalidReadingInvalidValueType.Event.ProfileName, invalidReadingInvalidValueType.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No SimpleReading Value", []requests.AddEventRequest{noSimpleValue}, noSimpleValue.Event.ProfileName, noSimpleValue.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No BinaryReading BinaryValue", []requests.AddEventRequest{noBinaryValue}, noBinaryValue.Event.ProfileName, noBinaryValue.Event.DeviceName, true, http.StatusBadRequest},
+		{"Invalid - No BinaryReading MediaType", []requests.AddEventRequest{noBinaryMediaType}, noBinaryMediaType.Event.ProfileName, noBinaryMediaType.Event.DeviceName, true, http.StatusBadRequest},
```

Comment:
Do we need tests where ProfileName or DeviceName don't match those in the Event?

Reply:
Well, the protocol layer will combine multiple Events inside a request, so it may not ideal to add unit test for validating mismatch profile/device name at protocol layer.  Instead, I added corresponding unit test at application layer against func `ValidateEvent`, and we may expand the unit test once `ValidateEvent` covers more validation against an event.